### PR TITLE
(HTCONDOR-2122)  Now with more C++20.

### PR DESCRIPTION
**Pull request by Todd-L-Miller**
This reverts commit dec9cef5f161c5c593c48a447cb2a85a53f071f0.

# HTCondor Pull Request Checklist for internal reviewers

- [ ] Verify that (GitHub thinks) the merge is clean. If it isn't, and you're confident you can resolve the conflicts, do so. Otherwise, send it back to the original developer.
- [ ] Verify that the related Jira ticket exists and has a target version number and that it is correct.
- [ ] Verify that the Jira ticket is in review status and is assigned to the reviewer.
- [ ] Verify that the Jira ticket (HTCONDOR-xxx) is mentioned at the beginning of the title. Edit it, if not
- [ ] Verify that the branch destination of the PR matches the target version of the ticket
- [ ] Check for correctness of change
- [ ] Check for regression test(s) of new features and bugfixes (if the feature doesn't require root)
- [ ] Check for documentation, if needed  (documentation [build logs](https://readthedocs.org/projects/htcondor/builds/))
- [ ] Check for version history, if needed
- [ ] Check BaTLab dashboard for successful build (https://batlab.chtc.wisc.edu/results/workspace.php) and test for either the PR or a workspace build by the developer that has the Jira ticket as a comment.
- [ ] Check that each commit message references the Jira ticket (HTCONDOR-xxx)

## After the above
- Hit the merge button if the pull request is approved and it is not a security patch (security changes require 2 additional reviews)
- If the pull request is approved, take the ticket out of review state
- Assign JIRA Ticket back to the developer

### CODE REVIEW

File: src/condor_starter.V6.1/filter.cpp [hunk 1]
```diff
@@ -4,105 +4,89 @@
...
-// Excepting the special case, these two function are identical, so
-// they should be refactored together.  Maybe by having a callback
-// for each CLASSAD_VALUE (with the extracted ClassAd and holding
-// attribute name as arguments)?
-//
-// ... would probably be less ugly as a two-pass algorithm in the
-// top-level wrapper.
-//
 
+typedef std::map<std::string, classad::Value::ValueType, classad::CaseIgnLTStr> Filter;
```
Comment (GregThain) [inline src/condor_starter.V6.1/filter.cpp:16]: I'm leaning towards the `using existing = new_type` syntax more and more over typedef, fwiw

File: src/condor_starter.V6.1/filter.cpp [hunk 2]
```diff
@@ -114,67 +98,42 @@ filterPluginResults( const classad::ClassAd & ad ) {
...
-            dprintf( D_NEVER, "failed to evaluate attribute %s\n", attribute.c_str() );
+        std::vector<ExprTree *> exprs;
+        list->GetComponents(exprs);
 
-            std::string buffer;
-            classad::ClassAdUnParser caup;
-            caup.Unparse(buffer, e);
-            dprintf( D_NEVER, "expression was '%s'\n", buffer.c_str() );
+        std::vector<ExprTree *> filteredExprs;
+        for( ExprTree * expr : exprs ) {
```
Comment (GregThain) [inline src/condor_starter.V6.1/filter.cpp:128]: can this be `for (const ExprTree *expr : exprs) {`

File: src/condor_starter.V6.1/filter.cpp [hunk 3]
```diff
@@ -114,67 +98,42 @@ filterPluginResults( const classad::ClassAd & ad ) {
...
+            classad::ClassAd * transferErrorDataAd = NULL;
```
Comment (GregThain) [inline src/condor_starter.V6.1/filter.cpp:129]: We should use `nullptr` everywhere in new code.

**Review by GregThain — CHANGES_REQUESTED** [review]

File: src/condor_starter.V6.1/filter.cpp (hunk 1 above)
Comment (Todd-L-Miller) [reply src/condor_starter.V6.1/filter.cpp:16]: What's the technical difference?  (If it's not just syntax?)

Comment (GregThain) [reply src/condor_starter.V6.1/filter.cpp:16]: Just syntax.

File: src/condor_starter.V6.1/filter.cpp (hunk 2 above)
Comment (Todd-L-Miller) [reply src/condor_starter.V6.1/filter.cpp:128]: Not immediately; `isClassad()` is both capitalized wrong and not marked `const`.  I'll try changing the latter and seeing what happens...

File: src/condor_starter.V6.1/filter.cpp (hunk 3 above)
Comment (Todd-L-Miller) [reply src/condor_starter.V6.1/filter.cpp]: Changed.

File: src/condor_starter.V6.1/filter.cpp (hunk 2 above)
Comment (Todd-L-Miller) [reply src/condor_starter.V6.1/filter.cpp]: ... looks like it Just Works(TM).  Changed.

**Review by GregThain — APPROVED** [review]
LGTM, subject to clean batlab build.

**Pull request merged**